financialaid/views.py

```diff
@@ -51,6 +56,35 @@ def get_queryset(self):  # pragma: no cover
         return None
 
 
+class FinancialAidSkipView(UpdateAPIView):
+    """
+    View for financial aid skip API. Takes user and program, then determines whether a financial
+    aid object exists, and then either creates or updates a financial aid object to reflect
+    the user skipping financial aid.
+    """
+    serializer_class = FinancialAidSkipSerializer
+    authentication_classes = (SessionAuthentication, )
+    permission_classes = (IsAuthenticated, )
+
+    def get_object(self):  # pragma: no cover
+        """
+        Overrides get_object in case financialaid object does not exist, as the learner may skip
+        financial aid either after starting the process or in lieu of applying
+        """
+        program = get_object_or_404(Program, id=self.kwargs["program_id"])
+        if not program.financial_aid_availability:
+            raise ValidationError("Financial aid not available for this program.")
+        if not ProgramEnrollment.objects.filter(program=program.id, user=self.request.user).exists():
+            raise ValidationError("User not in program.")
+        tier_program = get_no_discount_tier_program(program.id)
+        financialaid, _ = FinancialAid.objects.get_or_create(
+            user=self.request.user,
+            tier_program__program=program,
+            defaults={"tier_program": tier_program}
+        )
```

Comment:
Hmmm, I think this is one of those cases where we're implementing something sufficiently complex and non-standard that it won't really fit perfectly into REST.  I still think this is the right way to do it, given the intended functionality, but it's also likely to change in the future if there is going to be more functionality added to financial aid.

Why don't we leave it this way for now, and if either of us thinks of a better way to do this, we can bring it back up?  If you're still concerned about the partial update, we could also just override `patch()` and raise a 405/403/400?